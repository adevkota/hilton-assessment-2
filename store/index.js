
import { createStore } from "redux";
import { loadState } from "../util/persistHelper";

const defaultRoomState = {
	selected: false,
	numOfAdults: 1,
	numOfChildren: 0,
	selectable: true,
}
const defaultState = {
	rooms: [
		{ ...defaultRoomState, selected: true, selectable: false },
		{ ...defaultRoomState },
		{ ...defaultRoomState },
		{ ...defaultRoomState },
	],
	foo: 'ha',
}


const reducer = (state = defaultState, action) => {
	let rooms;
	switch (action.type) {
		case 'FOO':
			let savedRooms = action.payload.savedRooms;
			if (savedRooms) {
				rooms = state.rooms.map((room, index) => {
					if (savedRooms) {
						return {
							...savedRooms[index]
						};
					}
					return room;
				})

				return {
					...state,
					rooms
				}
			}
			return {
				...state
			};
		case 'ROOM_SELECTED':
			rooms = state.rooms.map((room, index) => {
				if (index > action.index) {
					return room;
				}
				return {
					...room,
					selected: true
				}
			});
			return {
				...state,
				rooms
			}
		case 'ROOM_UNSELECTED':
			rooms = state.rooms.map((room, index) => {
				if (index < action.index) {
					return room;
				}
				return {
					...defaultRoomState
				}
			});
			return {
				...state,
				rooms
			}
		case 'NUMBER_OF_ADULT_CHANGED':
			rooms = state.rooms.map((room, index) => {
				if (index !== action.index) {
					return room;
				}
				return {
					...room,
					numOfAdults: action.value
				}
			});
			return {
				...state,
				rooms
			}
		case 'NUMBER_OF_CHILDREN_CHANGED':
			rooms = state.rooms.map((room, index) => {
				if (index !== action.index) {
					return room;
				}
				return {
					...room,
					numOfChildren: action.value
				}
			});
			return {
				...state,
				rooms
			}
		default:
			return { ...state};
	}
};
export const makeStore = (initialState , options) => {
	return createStore(reducer, initialState);
}