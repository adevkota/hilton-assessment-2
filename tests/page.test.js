import configureMockStore from "redux-mock-store";
import { defaultStore } from "../store";
import { mount } from "enzyme";
import Page from "../components/page";
import { Provider } from "react-redux";
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

const mockStore = configureMockStore();
const store = mockStore(defaultStore);
const rooms = [{
	selected: false,
	numOfAdults: 1,
	numOfChildren: 0,
	selectable: true,
}]

test("renders without crashing", () => {
	const div = document.createElement("div");

	ReactDOM.render((
		<Provider store={store}>
			<Page
				rooms={rooms}
				roomSelected={()=>{}} 
				roomUnselected={()=>{}} 
				numOfAdultsChanged={()=>{}} 
				numOfChildrenChanged={()=>{}}
				loadFromCache={()=>{}}
			/>
		</Provider>
	), div);
	ReactDOM.unmountComponentAtNode(div);
});

test("matches snapshot for a selectable room", () => {
	const component = renderer.create(
		<Provider store={store}>
			<Page
				rooms={rooms}
				roomSelected={()=>{}} 
				roomUnselected={()=>{}} 
				numOfAdultsChanged={()=>{}} 
				numOfChildrenChanged={()=>{}}
				loadFromCache={()=>{}}
			/>
		</Provider>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});