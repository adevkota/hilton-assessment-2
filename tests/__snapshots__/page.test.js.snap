// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for a selectable room 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="page__Rooms-sc-1ls9xiw-0 pTDeI"
    >
      <div
        className="room__RoomWrapper-sc-14aj6pk-0 gDnqyN"
        selected={false}
      >
        <div
          className="room__RoomHeader-sc-14aj6pk-1 ekliE"
          selected={false}
        >
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span>
            Room 
            1
          </span>
        </div>
        <div
          className="room__RoomContent-sc-14aj6pk-2 cWkNCU"
          selected={false}
        >
          <div
            className="room__HeadCountWrapper-sc-14aj6pk-3 dNzWIe"
          >
            <span>
              Adults
            </span>
            <span>
              (18+)
            </span>
            <select
              className="adult"
              disabled={true}
              onChange={[Function]}
              value={1}
            >
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
            </select>
          </div>
          <div
            className="room__HeadCountWrapper-sc-14aj6pk-3 dNzWIe"
          >
            <span>
              Children
            </span>
            <span>
              (0-17)
            </span>
            <select
              className="children"
              disabled={true}
              onChange={[Function]}
              value={0}
            >
              <option
                value="0"
              >
                0
              </option>
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
    <button
      className="page__Button-sc-1ls9xiw-1 gcvhXZ"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
